spec/models/stamp_duty_spec.rb

```diff
@@ -24,57 +50,161 @@
     context 'when house price is 0' do
       let(:price) { 0 }
 
-      its(:tax_due) { is_expected.to be_zero }
-      its(:percentage_tax) { is_expected.to be_zero }
-      its(:total_due) { is_expected.to be_zero }
+      context 'and is not a second home' do
+        let(:second_home) { 'false' }
+
+        its(:tax_due) { is_expected.to be_zero }
+        its(:percentage_tax) { is_expected.to be_zero }
+        its(:total_due) { is_expected.to be_zero }
+      end
+
+      context 'and is a second home' do
+        let(:second_home) { 'true' }
+
+        its(:tax_due) { is_expected.to be_zero }
+        its(:percentage_tax) { is_expected.to be_zero }
+        its(:total_due) { is_expected.to be_zero }
+      end
+    end
+
+    context 'when house price is 40000' do
+      let(:price) { 40000 }
+
+      context 'and is not a second home' do
+        let(:second_home) { 'false' }
+
+        its(:tax_due) { is_expected.to be_zero }
+        its(:percentage_tax) { is_expected.to be_zero }
+        its(:total_due) { is_expected.to eq(40000) }
+      end
+
+      context 'and is a second home' do
+        let(:second_home) { 'true' }
+
+        its(:tax_due) { is_expected.to eq(1200) }
+        its(:percentage_tax) { is_expected.to eq(3) }
+        its(:total_due) { is_expected.to eq(41200) }
+      end
     end
 
     context 'when house price is 125000' do
       let(:price) { 125000 }
 
-      its(:tax_due) { is_expected.to be_zero }
-      its(:percentage_tax) { is_expected.to be_zero }
-      its(:total_due) { is_expected.to eql(125000) }
+      context 'and is not a second home' do
```

Comment:
Same rules apply. It's just down to whether or not you have the added checkbox ticked or not.